src/overlays/actors/ovl_En_Honotrap/z_en_honotrap.c

```diff
@@ -43,127 +85,727 @@ ActorInit En_Honotrap_InitVars = {
+    cos = Math_CosS(this->actor.home.rot.y);
+    Actor_SetScale(&this->actor, 0.1f);
+
+    if (sIsFirstInitEye) {
+        sIsFirstInitEye = false;
+        for (k = 0; k < HONOTRAP_EYE_MAX; k++) {
+            sSilverEyeTextures[k] = Lib_SegmentedToVirtual(sSilverEyeTextures[k]);
+        }
+    }
+
+    Collider_InitTris(play, &this->collider.tris);
+    Collider_SetTris(play, &this->collider.tris, &this->actor, &sTrisInit, this->collider.elements);
+
+    for (i = 0; i < 2; i++) {
+        for (j = 0, vtx = triangle; j < 3; j++, vtx++) {
+            Vec3f* baseVtx = &sTrisInit.elements[i].dim.vtx[j];
+
+            vtx->x = (baseVtx->z * sin) + (baseVtx->x * cos);
+            vtx->y = baseVtx->y;
+            vtx->z = (baseVtx->z * cos) - (baseVtx->x * sin);
+
+            Math_Vec3f_Sum(vtx, &this->actor.world.pos, vtx);
+        }
+        Collider_SetTrisVertices(&this->collider.tris, i, &triangle[0], &triangle[1], &triangle[2]);
+    }
+    Actor_SetFocus(&this->actor, 0.0f);
+    if (this->actor.params == HONOTRAP_TYPE_EYE) {
+        EnHonotrap_SetupEyeIdle(this);
+    } else { // HONOTRAP_TYPE_EYE_MUTI_FLAME
+        EnHonotrap_SetupEyeIdle2(this);
+    }
+}
+
+void EnHonotrap_InitFlame(EnHonotrap* this, PlayState* play) {
+    s32 pad[2];
+    s32 params = this->actor.params;
+
+    Actor_SetScale(&this->actor, 0.0001f);
+    Collider_InitCylinder(play, &this->collider.cyl);
+    Collider_SetCylinder(play, &this->collider.cyl, &this->actor, &sCylinderInit);
+    Collider_UpdateCylinder(&this->actor, &this->collider.cyl);
+    this->actor.terminalVelocity = -1.0f;
+    CollisionCheck_SetInfo(&this->actor.colChkInfo, NULL, &sColChkInfoInit);
+    ActorShape_Init(&this->actor.shape, 0.0f, ActorShadow_DrawCircle, 30.0f);
+    this->actor.shape.shadowAlpha = 128;
+    this->targetPos = GET_PLAYER(play)->actor.world.pos;
+    this->targetPos.y += 10.0f;
+    this->flameScroll = Rand_ZeroOne() * 511.0f;
+    Actor_PlaySfx(&this->actor, NA_SE_EV_FLAME_IGNITION);
+    if (params == HONOTRAP_TYPE_FLAME_DROP) {
+        this->actor.room = -1;
+        this->collider.cyl.dim.radius = 12;
+        this->collider.cyl.dim.height = 30;
+        this->actor.shape.yOffset = -1000.0f;
+    }
+    EnHonotrap_SetupFlameGrow(this);
+}
+
+void EnHonotrap_InitFlameGroup(EnHonotrap* this, PlayState* play) {
+    EnHonotrapFlameGroup* flameGroup;
+    s32 flameScroll;
+    s32 i;
+
+    Actor_SetScale(&this->actor, 0.0001f);
+    Collider_InitCylinder(play, &this->collider.cyl);
+    Collider_SetCylinder(play, &this->collider.cyl, &this->actor, &sCylinderInit);
+    CollisionCheck_SetInfo(&this->actor.colChkInfo, NULL, &sColChkInfoInit);
+
+    flameScroll = Rand_ZeroOne() * 511.0f;
+    flameGroup = &this->flameGroup;
+    for (i = 0; i < ARRAY_COUNT(flameGroup->flameList); i++) {
+        flameGroup->flameList[i].flameScroll = flameScroll;
+        flameScroll += (s32)Rand_ZeroFloat(300.0f) + 100;
+        flameScroll &= 0x1FF;
```

Comment:
If it matches
```suggestion
        flameScroll %= 0x200;
```

Reply:
not equivalent

Reply:
All three cases work if you add a `U` at the end of the mod: `flameScroll %= 0x200U;`
if you want